kong/plugins/apifortress/forward.lua

```diff
@@ -0,0 +1,96 @@
+-- ©2016 API Fortress Inc.
+-- Forwards the data to the configured API Fortress engine
+local bit = require "bit"
+local cjson = require "cjson"
+local basic_serializer = require "kong.plugins.log-serializers.basic"
+local serializer = require "kong.plugins.apifortress.fortress_serializer"
+local url = require "socket.url"
+
+local function parse_url(host_url)
+  local parsed_url = url.parse(host_url)
+  if not parsed_url.port then
+    if parsed_url.scheme == "http" then
+      parsed_url.port = 80
+     elseif parsed_url.scheme == HTTPS then
+      parsed_url.port = 443
+     end
+  end
+  if not parsed_url.path then
+    parsed_url.path = "/"
+  end
+  return parsed_url
+end
+
+local function generate_post_payload(parsed_url, key, token, message)
+  local body = cjson.encode(message)
+  local payload = string.format(
+    "%s %s?%s HTTP/1.1\r\nHost: %s\r\nConnection: Keep-Alive\r\nContent-Type: application/json\r\nX-Key: %s\r\nX-Token: %s\r\nContent-Length: %s\r\n\r\n%s",
+    "POST", parsed_url.path, parsed_url.query, parsed_url.host, key, token, string.len(body), body)
+  return payload
+end
+
+local function genToken(conf)
+	local timestamp = tostring(os.time())
```

Comment:
ngx_lua offers `ngx.time()` and `ngx.now()` which you can use at your convenience. `os.time()` on the other hand, executes a system call and cannot be JIT compiled and really hurts performance.